pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -423,15 +424,14 @@ func (rq *ResourceQuotaController) Sync(discoveryFunc NamespacedResourcesFunc, p
 			return
 		}
 
-		// Something has changed, so track the new state and perform a sync.
-		klog.V(2).Infof("syncing resource quota controller with updated resources from discovery: %v", newResources)
-		oldResources = newResources
-
 		// Ensure workers are paused to avoid processing events before informers
 		// have resynced.
 		rq.workerLock.Lock()
 		defer rq.workerLock.Unlock()
 
+		// Something has changed, so track the new state and perform a sync.
+		klog.V(2).Infof("syncing resource quota controller with updated resources from discovery: %s", printDiff(oldResources, newResources))
```

Comment:
Put this under `if klog.V(2) { klog.Infof .. }` so we don't pay the diff cost always.